client/components/seo/reader-preview/index.js

```diff
@@ -6,64 +6,30 @@ import React, { PropTypes, PureComponent } from 'react';
 /**
  * Internal dependencies
  */
-import humanDate from 'lib/human-date';
+import DisplayTypes from 'state/reader/posts/display-types';
+import ReaderPostCard from 'blocks/reader-post-card';
 
 export class ReaderPreview extends PureComponent {
 	render() {
 		const {
-			siteTitle,
-			siteSlug,
-			siteIcon,
-			postTitle,
+			site,
+			post,
 			postExcerpt,
 			postImage,
-			postDate,
-			authorName,
-			authorIcon
 		} = this.props;
 
+		const readerPost = { ...post };
+		if ( postImage ) {
+			readerPost.canonical_media = { src: postImage };
+		}
+		readerPost.better_excerpt = postExcerpt;
+
+		if ( postImage && ! postExcerpt ) {
+			readerPost.display_type = DisplayTypes.PHOTO_ONLY;
+		}
+
```

Comment:
fun fact! we can turn this into a single declarative statement with `Object.assign()`

```js
const readerPost = Object.assign(
	{ better_excerpt: postExcerpt },
	post,
	postImage && { canonical_media: { src: postImage } },
	( postImage && ! postExcerpt ) && { display_type: DisplayTypes.PHOTO_ONLY },
);
```